packages/mui-base/src/utils/useControlled.ts

```diff
@@ -0,0 +1,4 @@
+/**
+ * @ignore - internal hook.
+ */
+export { unstable_useControlled as useControlled } from '@mui/utils';
```

Comment:
Should it be imported from `@mui/utils/useControlled` to allow for tree shaking in dev mode? 
@mnajdova I would love for us to settle on a clear guideline of which packages should be tree-shakable in dev (and always have lint rules to forbid root imports), and clearly state that for the others we don't care.
Because right now we tend to try to have nested imports, but not always, which totally defeats the purpose.

Reply:
Thanks for nudging me. Is there a concern with always using nested imports? Agree, a lint rule could help, we likely have one if I am not mistaken. Even if it is not for tree shaking, we've seen this as an optimization for Pigment CSS too, as it needs to go through fewer files when the imports are nested.

Reply:
>  Is there a concern with always using nested imports?

For me the only problem is that if 19 persons always use nested imports for the 20th decide that it's not worth the effort and uses one root import, it can destroy the whole optilmization.
So I think having an explicit decision can help us enforce the rule.

But my personal opinion, I think having nested imports as a general rule for our codebase would be nice.

Reply:
Updated